design-proposals/golden-image-delivery-and-update-pipeline.md

```diff
@@ -0,0 +1,275 @@
+# Overview
+This design provides an approach for automating golden disk image importing and keeping golden disk images up-to-date. The design also outlines how to integrate with this automation across the KubeVirt ecosystem including install (HCO+SSP), the OCP UI, and Helm/OCP VM templates.
+
+## Motivation
+Iaas platforms (like AWS and GCP) provide root disk images for commonly used operating systems as well as continually providing updated versions of those images. By making commonly used disk images both automatically available and automatically up-to-date, this lowers the friction involved with using their platform.
+
+To close this usability gap between Iaas platforms and KubeVirt, the KubeVirt ecosystem should provide the tools necessary for supporting a similar pattern of disk image availability within the cluster.
+
+## Terms
+
+* Golden Disk Image: an immutable disk image stored in a PVC that can be cloned by a VM to be used as it's root OS image. Functionly, this concept is similar to an AWS AMI.
+* Golden Disk Image Namespace: A namespace that specifically exists to host golden disk image PVCs. VMs can use CDI to cross namespace clone PVCs from this golden disk image namespace anywhere else in the cluster.
+
+## Goals
+
+* Pipeline capable of importing and keeping golden images up-to-date
+* Zero operational downtime during disk updates
+* No mutating existing PVCs during the import/update flow
+* Compatible with DataVolumes
+* Compatible with Templates (meaning helm and ocp templates)
+* Transitions seamlessly with PVC Volume Populators in the future
+
+## Non Goals
+
+* Not a pipeline for updating disks attached to existing VMs
+
+## User Stories
+
+* As a vendor/distributor of OS images, I would like the most fresh version of my OS to reach the KubeVirt clusters that want it.
+* As a user creating VMs, I want commonly used operating system disk images to automatically be made available to me without requiring uploading and managing my own os boot disk images.
+* As a user creating VMs from a template, I want to be able to stamp out a new VM from a template which will always use the latest operating system version available.
+* As a cluster admin managing VM images, I want the ability to automate sourcing in operating system disk image updates and make them available to users.
```

Comment:
What about an alerting mechanism if there's a failure to import a new image?
Should there be an event on successful import?

Reply:
yep, good point.  I added some comments about monitoring and alerts to both the user stories and SSP operator responsibilities